src/modules/battery.rs

```diff
@@ -0,0 +1,77 @@
+use ansi_term::Color;
+
+use super::{Context, Module};
+
+/// Creates a segment for the battery percentage and charging state
+pub fn segment(_context: &Context) -> Option<Module> {
+    const BATTERY_FULL: &str = "•";
+    const BATTERY_CHARGING: &str = "⇡";
+    const BATTERY_DISCHARGING: &str = "⇣";
+    const BATTERY_THRESHOLD: f32 = 10.0;
+
+    let battery_status = get_battery_status()?;
+    let BatteryStatus { state, percentage } = battery_status;
+
+    if percentage > BATTERY_THRESHOLD {
+        log::debug!(
+            "Battery percentage is higher than threshold ({} > {})",
+            percentage,
+            BATTERY_THRESHOLD
+        );
+        return None;
+    }
+
+    // TODO: Set style based on percentage when threshold is modifiable
+    let mut module = Module::new("battery");
+    module.set_style(Color::Red.bold());
+    module.get_prefix().set_value("");
+
+    match state {
+        battery::State::Full => {
+            module.new_segment("full", BATTERY_FULL);
+        }
```

Comment:
not a blocking issue:

Maybe i'm missing something or probably more of a nitpick with the library but mac is saying i'm 100% but we are outputting 99. It feels like the `state: Full` isn't being used despite the name of the method above? 

![May-22-2019 11-37-58](https://user-images.githubusercontent.com/6132021/58188124-1158b980-7c86-11e9-83f2-c89b78b5f786.gif)

Reply:
It looks like this may be the culprit:
https://github.com/starship/starship/pull/63/files#diff-fb029b5925bdda2acb42ed4324807a7aR44

We are using `.trunc()` instead of `.round()`, so your `0.998...` will never be  100%. Nice find! 😄